packages/dds/tree/src/simple-tree/api/configuration.ts

```diff
@@ -262,9 +271,17 @@ function formatTypes(allowed: Iterable<TreeNodeSchema>): string {
 }
 
 /**
- * Detect cases documented in {@link ITreeConfigurationOptions.preventAmbiguity}.
+ * Check union contents are valid (shallowly).
+ * @remarks
+ * If preventAmbiguity is true, detect cases documented in {@link ITreeConfigurationOptions.preventAmbiguity}, reporting them to `ambiguityErrors`.
+ *
+ * Also checks for other invalid cases, such as duplicate schemas in the union.
  */
-export function checkUnion(union: Iterable<TreeNodeSchema>, errors: string[]): void {
+export function checkUnion(
+	union: Iterable<TreeNodeSchema>,
+	preventAmbiguity: boolean,
+	ambiguityErrors: string[],
```

Comment:
Nit: looks like this is mutated by the function = might be worth calling that out in the docs